src/components/calculator/calculator.js

```diff
@@ -1,197 +1,195 @@
-import React from 'react';
+import React, { useState } from 'react';
 import PropTypes from 'prop-types';
 
 import './calculator.css';
 import calculate from './logic/calculate';
 
+function prepareData(expresion) {
+  let { total, operation, next } = expresion;
+  total = total || '';
+  operation = operation || '';
+  next = next || '';
+  let result = total + operation + next;
+  result = result === '' ? 0 : result;
+  return result;
+}
 // eslint-disable-next-line react/prefer-stateless-function
-class Calculator extends React.Component {
-  constructor(props) {
-    super(props);
-    this.clickHandler = this.clickHandler.bind(this);
-    this.state = {
-      total: 0,
-      next: 0,
-      operation: null,
-    };
-  }
+const Calculator = (props) => {
+  const [expresion, setExpresion] = useState({
+    total: 0,
+    next: 0,
+    operation: null,
+  });
 
-  clickHandler(e) {
-    const { updateResult } = this.props;
+  const clickHandler = (e) => {
+    const { updateResult } = props;
     const btnName = e.target.innerText;
-    const newState = calculate(this.state, btnName);
-    this.setState(newState, () => {
-      let { total, operation, next } = this.state;
-      total = total || '';
-      operation = operation || '';
-      next = next || '';
-      let result = total + operation + next;
-      result = result === '' ? 0 : result;
+    const newExpresion = calculate(expresion, btnName);
+    setExpresion(newExpresion, () => {
+      const result = prepareData(newExpresion);
```

Comment:
- [ ] Good job so far but your events are not working. Kindly fix this console error and make sure your events work as expected 👍 
- Your app:
![Screenshot 2022-06-30 114954](https://user-images.githubusercontent.com/85113567/176660482-ef442971-55b6-4dcb-8fc2-3228dd3fe6ff.png)

Reply:
done 👍 
Thank you 😃